rotkehlchen/db/filtering.py

```diff
@@ -98,23 +98,25 @@ class DBETHTransactionJoinsFilter(DBFilter):
     should_join_events: bool = False
 
     def prepare(self) -> Tuple[List[str], List[Any]]:
-        query_filter, bindings = '', []
+        query_filters, bindings = [], []
         if self.should_join_events is True:
-            query_filter += (
-                ' INNER JOIN '
-                '(SELECT event_identifier, counterparty, asset FROM history_events) '
-                'ON hex(ethereum_transactions.tx_hash)=substr(event_identifier, 3)'
+            query_filters.append(
+                'LEFT JOIN (SELECT event_identifier, counterparty, asset FROM history_events) '
+                'ON hex(ethereum_transactions.tx_hash)=substr(event_identifier, 3)',
```

Comment:
Changed the approach of building a join query.

Now we use `LEFT JOIN ON` for `history_events` to include transactions that have no events and then we switch to `WHERE` condition.